### PR TITLE
Update DG with Command History and Command Assistant

### PR DESCRIPTION
**Pull request by Jiwei99**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -451,6 +453,72 @@ Step 10. The headers and rows are written to the CSV file that is specified by t
...
+
+### Command Assistant feature
+
+#### Implementation
+
+The operation is exposed in `SystemCommand`, specifically in `SystemCommand#execute`.
+
+The user keystroke will be read in `CommandBox#handleKeyStroke`. The user input will be evaluated by `SystemCommand#execute` to determine the appropriate help message to display.
+
+Given below is an example usage scenario and how the Find mechanism behaves at each step.
```
Comment (yucheng11122017) [inline docs/DeveloperGuide.md:498]: Should be command assistant feature

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -451,6 +453,72 @@ Step 10. The headers and rows are written to the CSV file that is specified by t
...
 
+### Command History feature
+
+#### Implementation
+
+The operation is exposed in `CommandBox` and `CommandHistory`.
+
+The user keystroke will be read in `CommandBox#handleKeyStroke`, which will call the appropriate method in `CommandHistory` to retrieve the previous and next commands. The retrieved commands are passed into `CommandBox#commandTextField` to be displayed.
+
+Given below is an example usage scenario and how the Find mechanism behaves at each step.
```
Comment (yucheng11122017) [inline docs/DeveloperGuide.md:464]: Should be command history feature

**Review by yucheng11122017 — APPROVED** [review]
LGTM except for minor changes. You not adding diagrams?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -451,6 +453,72 @@ Step 10. The headers and rows are written to the CSV file that is specified by t
...
+
+The user keystroke will be read in `CommandBox#handleKeyStroke`. The user input will be evaluated by `SystemCommand#execute` to determine the appropriate help message to display.
+
+Given below is an example usage scenario and how the Find mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. All contacts are displayed by default.
+
+Step 2. The user enters `find ` to search for a matching contact.
+
+Step 3. `CommandBox#handleKeyStroke` reads `find ` keystroke and calls `SystemCommand#execute`.
```
Comment (cookiedan42) [inline docs/DeveloperGuide.md:504]: is `find ` a keystroke or a String?
Perhaps the space is a keystroke but it reads the String

**Review by cookiedan42 — APPROVED** [review]

**Pull request merged**